webxr/render_state_update.https.html

```diff
@@ -0,0 +1,102 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/webxr_util.js"></script>
+<script src="resources/webxr_test_constants.js"></script>
+<canvas />
+
+<script>
+let fakeDeviceInitParams = TRACKED_IMMERSIVE_DEVICE;
+
+let testSessionEnded = function(session, fakeDeviceController, t) {
+  return new Promise((resolve, reject) => {
+      resolve(session.end().then(() => {
+          try {
+              session.updateRenderState({});
+              assert_unreached("updateRenderState should fail when created with an ended session");
+          } catch (err) {
+              assert_equals(err.name, 'InvalidStateError');
+          }
+      }));
+  });
+};
+
+
+let testBaseLayer = function(session, fakeDeviceController, t, sessionObjects) {
+  return new Promise((resolve, reject) => {
+    navigator.xr.test.simulateUserActivation(() => {
+      navigator.xr.requestSession('immersive-ar').then((tempSession) => {
```

Comment:
A correctly implemented UA will not let you create a second immersive session in the first place, this can only work for two inline sessions

Reply:
good point, will change it.

Reply:
Right, but you need to end the first session, no? I don't see that happening here.

Reply:
Well I might be missunderstanding the specs, but in the requestSession() algorithm in the step 5.4.8 just before resolving the promise it says:

>  If immersive is true
> Set the active immersive session to session, and set pending immersive session to false.

So there is no need to end the session to be allowed to request another one.

Reply:
Look further up, in step 4, "if immersive is true", substep 2:

> If pending immersive session is true or active immersive session is not null, reject promise with an "InvalidStateError" DOMException and return promise.

Reply:
> Look further up, in step 4, "if immersive is true", substep 2:
> 
> > If pending immersive session is true or active immersive session is not null, reject promise with an "InvalidStateError" DOMException and return promise.

I know. The thing is that I request the second session when the first session is already resolved. So, just before resolving, the first session sets "pending immersive session" to false, so when the second request arrives the step 4 substep 2 you mention does not apply.

Reply:
Right, when the session is resolved the pending immersive session is set to false, but the active immersive session is set to the new session, and that's also checked in 4.2.

The spec should not allow simultaneous immersive sessions to exist. That's why there's a single active immersive session field. If the spec is somehow allowing this, that's a bug (but I still fail to see how it does), but either way, there can only ever be a single immersive session.

Reply:
You're right. Actually we do check that in WebKit code, so there must be a bug somewhere else. Thanks for the review, I'll upload a new version after figuring out why it isn't failing.

Reply:
The code was OK. The problem was the test. The exception in both cases (requesting to immersive sessions or creating a base layer with a different sessions) is `InvalidStateError`, so the test was actually capturing the former instead of the latter.